Common/Core/RecoDecay.h

```diff
@@ -672,28 +667,10 @@ class RecoDecay
       return;
     }
     // If we are here, we have to follow the daughter tree.
-    //printf("getDaughters: ");
-    //for (int i = 0; i < stage; i++) // Indent to make the tree look nice.
-    //  printf(" ");
-    //printf("Stage %d: %d (PDG %d) -> %d-%d\n", stage, index, PDGParticle, indexDaughterFirst, indexDaughterLast);
     // Call itself to get daughters of daughters recursively.
     stage++;
-    // Get daughters of the first daughter.
-    if (indexDaughterFirst > -1) {
-      getDaughters(particlesMC, particlesMC.iteratorAt(indexDaughterFirst), list, arrPDGFinal, depthMax, stage);
-    }
-    // Get daughters of the daughters in between if any.
-    // Daughter indices are supposed to be consecutive and in increasing order.
-    // Reverse order means two daughters.
-    if (indexDaughterFirst > -1 && indexDaughterLast > -1) {
-      for (auto iD = indexDaughterFirst + 1; iD < indexDaughterLast; ++iD) {
-        getDaughters(particlesMC, particlesMC.iteratorAt(iD), list, arrPDGFinal, depthMax, stage);
-      }
-    }
-    // Get daughters of the last daughter if different from the first one.
-    // Same indices indicate a single daughter.
-    if (indexDaughterLast > -1 && indexDaughterLast != indexDaughterFirst) {
-      getDaughters(particlesMC, particlesMC.iteratorAt(indexDaughterLast), list, arrPDGFinal, depthMax, stage);
+    for (auto& dauIdx : particle.daughtersIds()) {
+      getDaughters(particlesMC, particlesMC.rawIteratorAt(dauIdx), list, arrPDGFinal, depthMax, stage);
```

Comment:
This is the only place where the `particlesMC` table is used. Could it be removed by looping over `particle.daughters()` instead?

Reply:
Right! I removed the argument and I replaced the loop over indices with a loop over particles directly